### PR TITLE
API fixes and frontend/backend fixes

### LINKED ISSUE #10: Test frontend-backend communication
Test that communication between the frontend and backend works properly and that there are no formatting issues, etc

### LINKED ISSUE #12: Corrections to REST API and Tests
Update documentation 
Return 200 OK for completed POST/DELETE requests
Fix REST API tests

### PR DESCRIPTION
**Pull request by SteveJones92**
Fixes the API to take raw JSON format and fixes the tests to match.

closes #10
closes #12

### CODE REVIEW

File: src/wishlist.py [hunk 1]
```diff
@@ -39,14 +40,15 @@ def wishlist():
...
+    data = request.get_json(force=True)
+    user = data.get('user')
+    item = json.loads(data.get('item'))
 
     # TODO convert price to generic currency and fix price hack used below
 
     db.query(
         'INSERT INTO wishlist (username, name, price, website, link, rating) VALUES(%s, %s, %s, %s, %s, %s)',
-        (user, item['title'], float(item['price'][1:]), item['website'], item['link'], item['rating'])
+        (user, item['title'], float(item['price']), item['website'], item['link'], item['rating'])
```
Comment (alex-snezhko) [inline src/wishlist.py:51]: Right now the price is sent as a string including the $ sign. Could you look into making it store a 'universal' currency in the database and convert to appropriate currency when being sent to frontend and when taken from frontend?

File: src/search.py [hunk 2]
```diff
@@ -24,11 +24,11 @@
...
 @search_bp.route('/search/', methods=['GET'])
 def search():
-    results = scraper.driver(request.args.get('name'), request.args.get('currency'),
+    results = scraper.driver(request.args.get('title'), request.args.get('currency'),
                              int(request.args.get('numProducts')))
 
     # if not descending, it is ascending
     results = result_formatter.sortList(results, request.args.get('sortBy'),
-        request.args.get('displayOrder').lower() == 'desc')
+        True if request.args.get('displayOrder').lower() == 'desc' else False )
```
Comment (alex-snezhko) [inline src/search.py]: replace this with just `request.args.get('displayOrder').lower() == 'desc'`. The True if x else False is unnecessary and pylint complains about it

Comment (SteveJones92) [reply src/search.py]: True, easy fix, definitely overlooked that one

File: src/wishlist.py (hunk 1 above)
Comment (SteveJones92) [reply src/wishlist.py:51]: Assumed taking a number only for price, cut the inputs in the endpoint call down to a float from frontend would work as well, all of the tests are setup already for not sending in a $ so probably an quicker fix just to not send it

**Pull request merged**